shadowsocks/stream.go

```diff
@@ -262,7 +268,10 @@ type chunkReader struct {
 	aead cipher.AEAD
 	// Index of the next encrypted chunk to read.
 	counter []byte
-	buf     []byte
+	// Buffer for the uint16 size and its AEAD tag.  Made in init().
+	size []byte
```

Comment:
Why do we need two buffers? You can call Acquire before reading the size. Wouldn't it be simpler if we just had a single buffer as before? I find it a bit easier to reason about because it's the buffer for the output and it all goes together in one write.

Reply:
We need two buffers because idle streams sit blocked in the first `Read` call for up to minutes at a time, waiting for the next chunk.  During that blocked call, we don't want to be holding the full size buffer.